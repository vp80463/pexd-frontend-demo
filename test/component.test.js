import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import AddUser from '../src/views/demo/AddUser.vue';

describe('component of LoginPage.vue', () => {
  it('should render', () => {
    const wrapper = mount(AddUser);
    expect(wrapper.text()).toContain('Username');
    expect(wrapper.text()).toContain('Password');

    // to check the component html is match snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be interactive', async () => {
    const wrapper = mount(AddUser);
    expect(wrapper.find('.username').exists()).toBe(true);
    expect(wrapper.find('.password').exists()).toBe(true);

    const usernameInput = wrapper.find('.username .vue-input__inner');
    await usernameInput.setValue('admin');

    const passwordInput = wrapper.find('.password .vue-input__inner');
    await passwordInput.setValue('123456');

    expect(wrapper.vm.form.username).toBe('admin');
    expect(wrapper.vm.form.password).toBe('123456');

    await wrapper.get('button').trigger('click');
  });
});
