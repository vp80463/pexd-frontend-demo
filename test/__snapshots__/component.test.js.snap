// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component of LoginPage.vue > should render 1`] = `
"<div>
  <form class="vue-form vue-form--default vue-form--label-right" style="max-width: 400px; margin-bottom: 20px;">
    <div class="vue-form-item asterisk-left" role="group" aria-labelledby="vue-id-0">
      <div id="vue-id-0" class="vue-form-item__label" style="width: 80px;">Username</div>
      <div class="vue-form-item__content">
        <div class="vue-input username">
          <!---->
          <div class="vue-input__wrapper">
            <!----><input class="vue-input__inner" type="text" autocomplete="off" tabindex="0">
            <!---->
          </div>
          <!---->
        </div>
        <transition-group-stub name="vue-zoom-in-top" appear="false" persisted="false" css="true">
          <!---->
        </transition-group-stub>
      </div>
    </div>
    <div class="vue-form-item asterisk-left" role="group" aria-labelledby="vue-id-1">
      <div id="vue-id-1" class="vue-form-item__label" style="width: 80px;">Password</div>
      <div class="vue-form-item__content">
        <div class="vue-input password">
          <!---->
          <div class="vue-input__wrapper">
            <!----><input class="vue-input__inner" type="password" autocomplete="off" tabindex="0">
            <!---->
          </div>
          <!---->
        </div>
        <transition-group-stub name="vue-zoom-in-top" appear="false" persisted="false" css="true">
          <!---->
        </transition-group-stub>
      </div>
    </div>
    <div class="vue-form-item asterisk-left">
      <div class="vue-form-item__content" style="margin-left: 80px;"><button aria-disabled="false" type="button" class="vue-button vue-button--primary">
          <!----><span class=""> Add User </span>
        </button>
        <transition-group-stub name="vue-zoom-in-top" appear="false" persisted="false" css="true">
          <!---->
        </transition-group-stub>
      </div>
    </div>
  </form>
</div>"
`;
